File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
   public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        map<string, int> idx;
        int cur = 0;
        for (int i = 0; i < strs.size(); i++) {
            string temp = strs[i];
            sort(temp.begin(), temp.end());
            if (idx[temp] == 0) {
                ans.push_back({strs[i]});
                cur++;
                idx[temp] = cur;
            } else {
                ans[idx[temp] - 1].push_back(strs[i]);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.java
class Solution {
    public int characterReplacement(String s, int k) {
        int ans=0;
        int cur=k;
        int st=0;
       for(char i='A';i<='Z';i++)
        {
            st=0;
            cur=k;
            for(int j=0;j<s.length();j++)
            {
                char c=s.charAt(j);
                if(c!=i)
                {
                    cur--;
                }
                while(cur<0)
                {
                    char ch=s.charAt(st);
                    if(ch!=i)
                    cur++;
                    st++;
                }
                ans=Math.max(ans,j-st+1);
            }
        }
        return ans;
    }
}
